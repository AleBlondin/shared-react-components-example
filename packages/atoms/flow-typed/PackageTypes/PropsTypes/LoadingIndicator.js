// @flow

declare type LoadingIndicatorPropsType = {|
  className?: string,
  centered?: boolean,
  color?: string,
  loaderSize?: number,
|};
