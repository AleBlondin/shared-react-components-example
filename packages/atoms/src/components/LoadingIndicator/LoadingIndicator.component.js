// @flow

import React from 'react';
import { Loader } from 'react-loaders';
import 'loaders.css/loaders.min.css';

import { SharedComponentsColors } from 'shared-components-atoms/style';

import Style from './LoadingIndicator.style';

const LoadingIndicator = (props: LoadingIndicatorPropsType) => (
  <Style {...props}>
    <Loader color={props.color} type="ball-spin-fade-loader" />
  </Style>
);

LoadingIndicator.defaultProps = {
  className: null,
  color: SharedComponentsColors.brandGreen,
  loaderSize: 70,
};

export default LoadingIndicator;
