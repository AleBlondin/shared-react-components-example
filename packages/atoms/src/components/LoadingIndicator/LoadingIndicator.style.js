// @flow

import styled, { css } from 'styled-components';

const OFFSET_CENTER_PX = 10;
const SCALE_RATIO = 0.6;

const LoadingIndicatorStyle = styled.div`
  ${({ centered }) =>
    centered
      ? css`
          position: absolute;
          top: 50%;
          left: 50%;
          transform: translate(-50%, -50%);
        `
      : css`
          position: relative;
        `};

  .loader {
    ${({ loaderSize }) =>
      css`
        height: ${loaderSize}px;
        width: ${loaderSize}px;
        transform: scale(${SCALE_RATIO});
        position: relative;
      `};
  }

  .loader-inner {
    ${({ loaderSize }) =>
      css`
        left: ${loaderSize / 2 - OFFSET_CENTER_PX}px;
        top: ${loaderSize / 2 - OFFSET_CENTER_PX}px;
      `};
  }
`;

export default LoadingIndicatorStyle;
