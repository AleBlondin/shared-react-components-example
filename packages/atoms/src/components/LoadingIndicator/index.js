// @flow

export { default } from './LoadingIndicator.component';
