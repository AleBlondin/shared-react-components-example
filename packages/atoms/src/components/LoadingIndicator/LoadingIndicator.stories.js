import React from 'react';

// eslint-disable-next-line import/no-extraneous-dependencies
import { storiesOf } from '@storybook/react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { boolean, withKnobs } from '@storybook/addon-knobs';

import LoadingIndicator from './LoadingIndicator.component';

storiesOf('Atoms', module)
  .addDecorator(withKnobs)
  .add('LoadingIndicator', () => (
    <LoadingIndicator centered={boolean('Centered', false)} />
  ));
