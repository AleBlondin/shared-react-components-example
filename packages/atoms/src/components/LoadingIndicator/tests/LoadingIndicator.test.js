import React from 'react';
import renderer from 'react-test-renderer';

import LoadingIndicator from '../LoadingIndicator.component';

describe('<LoadingIndicator />', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<LoadingIndicator />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
