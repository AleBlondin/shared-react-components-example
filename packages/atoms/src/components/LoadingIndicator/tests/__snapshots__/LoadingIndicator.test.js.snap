// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should match snapshot 1`] = `
.c0 {
  position: relative;
}

.c0 .loader {
  height: 70px;
  width: 70px;
  -webkit-transform: scale(0.6);
  -ms-transform: scale(0.6);
  transform: scale(0.6);
  position: relative;
}

.c0 .loader-inner {
  left: 25px;
  top: 25px;
}

<div
  className="c0"
  color="#00965E"
>
  <div
    className="loader loader-undefined loader-active"
    style={undefined}
  >
    <div
      className="loader-inner ball-spin-fade-loader"
    >
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
      <div
        style={
          Object {
            "backgroundColor": "#00965E",
          }
        }
      />
    </div>
  </div>
</div>
`;
